compiler/include/expr.h

```diff
@@ -247,6 +247,11 @@ class IfcConstraint final : public Expr {
   InterfaceSymbol* ifcSymbol()  const;
   int              numActuals() const { return consActuals.length; }
 
+  // true for constraints that are satisfied automatically from existing
+  // procedures, such as Hashable. Long-term, these should only use
+  // compiler-generated procedures, and not allow user-supplied ones.
+  bool shouldBeGeneratedOnly = false;
```

Comment:
See my comments on `chpl__defaultHashWrapper` but I don't think we need this; we can handle the warning and fall back on the old resolution by modifying the module code

Reply:
I think we do -- even if `chpl__defaultHashWrapper` has code to warn, what if a user tries to write their own `Hashable`-backed constrained-generic function:

```Chapel
proc f(x: Hashable) {
  // do not use chpl__defaultHashWrapper
}

record R {
  proc hash(): uint {
    return 0;
  }
}

f(new R());
```

The warning in this code would fire at this point and let them know that they're relying on the auto-implementation behavior. I've designed the "implements" primitive with this in mind, so that both warnings aren't issued together.

Reply:
Based on the separate conversation about buildDefaultFunctions, it's OK with me to keep this.